eth/handler.go

```diff
@@ -593,6 +608,25 @@ func (h *handler) txBroadcastLoop() {
 	}
 }
 
+func (h *handler) l2MigrationLoop() {
+	for {
+		select {
+		case event := <-h.newChainHeadCh:
+			block := event.Block
+			if h.chain.Config().IsL2Migration(block.Number()) {
+				log.Info("L2 Migration Block Reached, stopping handler and p2p server", "block", block.NumberU64(), "hash", block.Hash())
+				h.wg.Done()
+				h.Stop()
+				h.server.Stop()
+				return
+			}
+		case <-h.newChainHeadSub.Err():
+			h.wg.Done()
```

Comment:
Nitpick I randomly noticed: you could put it in a `defer` before the loop instead of all the return cases so that you don't forget to call this when a new return case gets added.

Reply:
Yea good catch and thanks for taking a look! This was actually done on purpose after I discovered that if we use defer for `h.wg.Done()` like in the other handler loops then we get a deadlock [here](https://github.com/celo-org/celo-blockchain/blob/e0fc697ca4f2cfe04618d9b337477c31bfe31f37/eth/handler.go#L494) inside `h.Stop()` because the defer statement hasn't been called yet. I'll add a comment to explain this